docs/docs-beta/docs/guides/asset-checks.md

```diff
@@ -1,64 +1,74 @@
+
+:::tip
+Dagster's dbt integration can model existing dbt tests as asset checks. Refer to the [dagster-dbt documentaiton](/integrations/dbt) for more information.
+:::
 
-The example below defines a single asset check on an asset that fails if the `order_id` column of the asset contains a null value.
+Single asset checks are defined using the `@asset_check` decorator.
 
-In this example, the asset check will run after the asset has been materialized, to ensure the quality of its data.
+The following example defines an asset check on an asset that fails if the `order_id` column of the asset contains a null value. The asset check will run after the asset has been materialized, to ensure the quality of its data.
 
-<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/single-asset-check.py" language="python" title="Asset with a single asset check" />
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/single-asset-check.py" language="python" />
 
-## Testing assets with multiple asset checks
+## Defining multiple asset checks \{#multiple-checks}
 
 In most cases, checking the data quality of an asset will require multiple checks.
 
-The example below defines two asset checks using the `@multi_asset_check` decorator:
+The following example defines two asset checks using the `@multi_asset_check` decorator:
 
 - One check that fails if the `order_id` column of the asset contains a null value
 - Another check that fails if the `item_id` column of the asset contains a null value
 
 In this example, both asset checks will run in a single operation after the asset has been materialized.
 
-<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/multiple-asset-checks.py" language="python" title="Asset with multiple asset checks" />
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/multiple-asset-checks.py" language="python" />
 
-## Programmatically generating asset checks
+## Programmatically generating asset checks \{#factory-pattern}
 
 Defining multiple checks can also be done using a factory pattern. The example below defines the same two asset checks as in the previous example, but this time using a factory pattern and the `@multi_asset_check` decorator.
 
-<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/asset-checks-factory.py" language="python" title="Defining asset checks using a factory pattern" />
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/asset-checks-factory.py" language="python" />
 
-## Blocking downstream assets
+## Blocking downstream materialization
 
-By default, if a parent's asset check fails during a run, the run will continue and downstream assets will be materialized. To prevent this behavior, set the `blocking` argument to `True` in the `@asset_check` decorator. This will cause the run to fail and prevent further materializations.
+:::info
+This feature is only supported by the `@asset_check` decorator.
```

Comment:
This seems like a pretty significant limitation.

My first questions is how to prevent materialization if one-or-the-other check fails. Does this fall under declarative automation?

Reply:
This is also not true in dbt, since dbt runs stop if a test fails

Reply:
Hmm, I'm not totally sure. DC can skip materializing downstream assets with failing/missing/in progress upstream parents, but AFAIK not due to something like null values. Maybe a question for @owen?